### PR TITLE
SCRIPTS/SQL: SMN Mods and Dream Shroud

### PR DESCRIPTION
**Pull request by Zenny89**

### CODE REVIEW

File: scripts/globals/abilities/pets/dream_shroud.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+end;
+
+function onPetAbility(target, pet, skill, summoner)
+    local bonusTime = utils.clamp(summoner:getSkillLevel(SKILL_SUM) - 300, 0, 200);
+    local duration = 180 + bonusTime;
+    local hour = VanadielHour();
+    local buffvalue = 0;
+	
+	if hour == 0 then
+	     buffvalue = 13;
```
Comment (TeoTwawki) [inline scripts/globals/abilities/pets/dream_shroud.lua]: Got mixed tabs and spaces in file. ~~Pick one please.~~ DSP wants spaces. 4 spaces = 1 indent.

**Comment by Zenny89** [comment]
Updated - I think that did it.

File: scripts/globals/abilities/pets/dream_shroud.lua [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+end;
+
+function onPetAbility(target, pet, skill, summoner)
+    local bonusTime = utils.clamp(summoner:getSkillLevel(SKILL_SUM) - 300, 0, 200);
+    local duration = 180 + bonusTime;
+    local hour = VanadielHour();
+    local buffvalue = 0;
+
+    if hour == 0 then
+             buffvalue = 13;
```
Comment (TeoTwawki) [inline scripts/globals/abilities/pets/dream_shroud.lua]: indent is still wrong. 2 indents should be 8 spaces. you have 13

Comment (TeoTwawki) [inline scripts/globals/abilities/pets/dream_shroud.lua]: Configure your editor to use 4 spaces in place of the tab key, so you don't have to count them out.

**Comment by Zenny89** [comment]
Corrected.

File: scripts/globals/abilities/pets/dream_shroud.lua [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    return 0,0;
+end;
+
+function onPetAbility(target, pet, skill, summoner)
+    local bonusTime = utils.clamp(summoner:getSkillLevel(SKILL_SUM) - 300, 0, 200);
+    local duration = 180 + bonusTime;
+    local hour = VanadielHour();
+    local buffvalue = 0;
+
+    if hour == 0 then
```
Comment (teschnei) [inline scripts/globals/abilities/pets/dream_shroud.lua]: buffvalue = math.abs(12 - hour) + 1 can replace this whole if/else chain

**Comment by Zenny89** [comment]
Updated

**Comment by TeoTwawki** [comment]
````lua
    local buffvalue = 0;
    buffvalue = math.abs(12 - hour) + 1
```

Uh, shouldn't that have just been

```lua
local buffvalue = math.abs(12 - hour) + 1
```

**Comment by TeoTwawki** [comment]
eh, oh well.

**Pull request merged**